pkg/deployment/depresource/start.go

```diff
@@ -0,0 +1,164 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package depresource
+
+import (
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/client/deployments"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/ecctl/pkg/deployment"
+	"github.com/elastic/ecctl/pkg/deployment/deputil"
+	"github.com/elastic/ecctl/pkg/util"
+)
+
+// StartParams is consumed by start.
+type StartParams struct {
+	*api.API
+	DeploymentID string
+	Type         string
+	RefID        string
+	All          bool
+}
+
+// Validate ensures the parameters are usable by Start.
+func (params StartParams) Validate() error {
+	var merr = new(multierror.Error)
```

Comment:
I think we have a common structure right @marclop to avoid duplication?

Reply:
Do you mean `depresource.NewStateless{}` ? I don't think that struct works very well here, It has a bunch of fields that aren't relevant, and also the `Validate()` and `fillDefaults()` methods are different.

I can make a common struct for these [commands](https://github.com/elastic/ecctl/pull/89/files#diff-037fd59d86ef9ef3885b80d1b6126216R33) and [`stop`/`stop-maintenance`](https://github.com/elastic/ecctl/pull/81/files#diff-ccddd9d334f1b421e58829b51a7593ccR33) called `StopStartParams` if you want.

Reply:
No , no , I can't find a discussion I had with Marc last week on a GH pr about the same. @marclop can u help pls? :)

Reply:
We discussed out of band that going forward probably a common structure with: `API`, `DeploymentID`, `RefID`, `Type` can be defined and used with all the validations in common. Including the fillDefaults ref-id auto-discovery.

Reply:
Ok so there will be another PR introducing the common struct with validation + filldefaults. 
Great. we are good to go then